src/nex_types.rs

```diff
@@ -0,0 +1,446 @@
+            sid: None,
+            cid: None,
+            pid: None,
+            transport_type: None,
+            rvcid: None,
+            natm: None,
+            natf: None,
+            upnp: None,
+            pmp: None,
+            probe_init: None,
+            prid: None
+        }
+    }
+}
+
+impl StationURL {
+    pub fn set_scheme(&mut self, scheme: String) {
+        self.scheme = Some(scheme);
+    }
+
+    pub fn set_address(&mut self, address: String) {
+        self.address = Some(address);
+    }
+
+    pub fn set_port(&mut self, port: String) {
+        self.port = Some(port);
+    }
+
+    pub fn set_stream(&mut self, stream: String) {
+        self.stream = Some(stream);
+    }
+
+    pub fn set_sid(&mut self, sid: String) {
+        self.sid = Some(sid);
+    }
+
+    pub fn set_cid(&mut self, cid: String) {
+        self.cid = Some(cid);
+    }
+
+    pub fn set_pid(&mut self, pid: String) {
+        self.pid = Some(pid);
+    }
+
+    pub fn set_transport_type(&mut self, transport_type: String) {
+        self.transport_type = Some(transport_type);
+    }
+
+    pub fn set_rvcid(&mut self, rvcid: String) {
+        self.rvcid = Some(rvcid);
+    }
+
+    pub fn set_natm(&mut self, natm: String) {
+        self.natm = Some(natm);
+    }
+
+    pub fn set_natf(&mut self, natf: String) {
+        self.natf = Some(natf);
+    }
+
+    pub fn set_upnp(&mut self, upnp: String) {
+        self.upnp = Some(upnp);
+    }
+
+    pub fn set_pmp(&mut self, pmp: String) {
+        self.pmp = Some(pmp);
+    }
+
+    pub fn set_probe_init(&mut self, probe_init: String) {
+        self.probe_init = Some(probe_init);
+    }
+
+    pub fn set_prid(&mut self, prid: String) {
+        self.prid = Some(prid);
+    }
+
+    pub fn get_scheme(self) -> Option<String> {
+        self.scheme
+    }
+
+    pub fn get_address(self) -> Option<String> {
+        self.address
+    }
+
+    pub fn get_port(self) -> Option<String> {
+        self.port
+    }
+
+    pub fn get_stream(self) -> Option<String> {
+        self.stream
+    }
+
+    pub fn get_sid(self) -> Option<String> {
+        self.sid
+    }
+
+    pub fn get_cid(self) -> Option<String> {
+        self.cid
+    }
+
+    pub fn get_pid(self) -> Option<String> {
+        self.pid
+    }
+
+    pub fn get_transport_type(self) -> Option<String> {
+        self.transport_type
+    }
+
+    pub fn get_rvcid(self) -> Option<String> {
+        self.rvcid
+    }
+
+    pub fn get_natm(self) -> Option<String> {
+        self.natm
+    }
+
+    pub fn get_natf(self) -> Option<String> {
+        self.natf
+    }
+
+    pub fn get_upnp(self) -> Option<String> {
+        self.upnp
+    }
+
+    pub fn get_pmp(self) -> Option<String> {
+        self.pmp
+    }
+
+    pub fn get_probe_init(self) -> Option<String> {
+        self.probe_init
+    }
+
+    pub fn get_prid(self) -> Option<String> {
+        self.prid
+    }
+
+    pub fn new_from_string(str: String) -> Self {
```

Comment:
If you run `cargo clippy`, it has a few small lint suggestions for this section.